pkg/controller/elasticsearch/sset/validation.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package sset
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PodTemplateError struct {
+	Parent      metav1.Object
+	StatefulSet appsv1.StatefulSet
+	Causes      []metav1.StatusCause
+}
+
+func (e *PodTemplateError) Error() string {
+	return fmt.Sprintf(
+		"Validation of PodTemplate for StatefulSet %s in Elasticsearch %s/%s failed for the following reasons: %v",
+		e.StatefulSet.Name,
+		e.Parent.GetNamespace(),
+		e.Parent.GetName(),
+		e.Causes,
+	)
+}
+
+// validatePodTemplate validates a Pod Template by issuing a dry API request.
```

Comment:
dry -> dry-run?

Reply:
`// This check is performed as "best-effort", since the dry-run API is only available starting K8s 1.15.0`